### PR TITLE
Detect Dark Mode Settings

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
With this PR the Dashboard can detect the system settings for dark mode

Screenshot 2021-03-22 at 13 26 03

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
Dark Mode: The Dashboard now applies system settings by default
```

### CODE REVIEW

File: frontend/src/components/MainToolbar.vue [hunk 1]
```diff
@@ -104,8 +104,17 @@ SPDX-License-Identifier: Apache-2.0
           <v-divider></v-divider>
           <v-card-actions class="px-3">
             <v-icon color="primary" class="ml-2 mr-3">mdi-brightness-6</v-icon>
-            <v-switch v-model="darkMode" dense hide-details class="ma-0" color="primary" @click.native.stop></v-switch>
-            <span class="primary--text text-button">Dark</span>
+            <v-btn-toggle v-model="darkMode" borderless mandatory @click.native.stop>
```
Comment (petersutter) [inline frontend/src/components/MainToolbar.vue:107]: add tooltips

Comment (grolu) [reply frontend/src/components/MainToolbar.vue:107]: ah yes I forgot that actually I wanted to do this

File: frontend/src/store/index.js [hunk 2]
```diff
@@ -1498,9 +1498,25 @@ const mutations = {
     state.splitpaneResize = value
   },
   SET_DARK_MODE (state, value) {
-    state.darkMode = value
-    localStorage.setItem('global/dark-mode', value)
-    Vue.vuetify.framework.theme.dark = value
+    if (value !== undefined) {
```
Comment (holgerkoser) [inline frontend/src/store/index.js:1501]: From my point of view this is an antipattern. A mutation should only mutate the store state and not run mediaqueries.  The new value must come from outside the mutation. Why do we need two properties `darkMode` and `darkTheme`.

Comment (holgerkoser) [reply frontend/src/store/index.js:1501]: We should start using store plugins for this kind of problems.
```js
export default function createMediaPlugin (window) {
  return store => {
    const mq = window.matchMedia('(prefers-color-scheme: dark)')
    store.commit('SET_PREFERS_COLOR_SCHEME', mq.matches ? 'dark' : 'light')
    mq.addListener(e => {
      store.commit('SET_PREFERS_COLOR_SCHEME', e.matches ? 'dark' : 'light')
    })
  }
}
```

File: frontend/src/app.js [hunk 3]
```diff
@@ -21,8 +21,13 @@ const App = Vue.extend({
       }
     })
 
-    const darkMode = this.$localStorage.getItem('global/dark-mode') === 'true'
+    const darkMode = this.$localStorage.getItem('global/setting-dark-mode') || 2
```
Comment (holgerkoser) [inline frontend/src/app.js:24]: Why `global/seeting-dark-mode` and not `global/dark-mode` or `global/settings/dark-mode`?

Comment (holgerkoser) [reply frontend/src/app.js:24]: I would propose to have a localStorage porperty 'global/color-scheme' or  'global/settings/color-scheme' with possible values `dark`, `light` and `auto` or `no-preference`

File: frontend/src/components/MainToolbar.vue (hunk 1 above)
Comment (grolu) [reply frontend/src/components/MainToolbar.vue:107]: done

File: frontend/src/components/MainToolbar.vue [hunk 4]
```diff
@@ -92,6 +92,32 @@ SPDX-License-Identifier: Apache-2.0
...
+                  </template>
+                  <span>Dark Mode</span>
+                </v-tooltip>
+                <v-tooltip top>
+                  <template v-slot:activator="{ on }">
+                    <v-btn small v-on="on">
+                      <v-icon color="primary">mdi-brightness-auto</v-icon>
+                    </v-btn>
+                  </template>
+                  <span>Auto</span>
```
Comment (petersutter) [inline frontend/src/components/MainToolbar.vue:118]:
```suggestion
                  <span>Automatically choose theme based on your system settings</span>
```

File: frontend/src/components/MainToolbar.vue [hunk 5]
```diff
@@ -199,12 +220,29 @@ export default {
...
+          case 0:
+            this.setColorScheme('light')
+            break
+          case 1:
+            this.setColorScheme('dark')
+            break
+          default:
+            this.setColorScheme('auto')
+            break
+        }
```
Comment (petersutter) [inline frontend/src/components/MainToolbar.vue:245]: use constants instead of magic/index numbers

Comment (grolu) [reply frontend/src/components/MainToolbar.vue]: done

**Review by petersutter — APPROVED** [review]
/lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**